server/src/index.ts

```diff
@@ -13,82 +14,119 @@ if (!fs.existsSync(uploadDir)) {
 }
 
 const ENDPOINTS = {
-  deleteEmployee: "/api/delete-employee",
-  addEmployee: "/api/add-employee",
-  getEmployees: "/api/employees",
+  employee: "/api/employee",
+  employees: "/api/employees",
 };
 
-const validateEmployeeObject = (obj: EmployeeI): boolean => {
+const validateEmployeeObject = (obj: IEmployee): boolean => {
   return !!obj?.name && !!obj?.lastName && !!obj?.occupation && !!obj?.age;
 };
 
-const getAllFiles = (): Array<EmployeeI> => {
-  const data: Array<EmployeeI> = [];
-  fs.readdirSync(uploadDir).forEach(file => {
-    const content = fs.readFileSync(`${uploadDir}${file}`);
-    data.push(JSON.parse(content.toString()) as EmployeeI);
-  });
+const parseAllFiles = async (): Promise<Array<IEmployee>> => {
+  const data: Array<IEmployee> = [];
+  const files = await readdir(uploadDir);
+  for (const file of files) {
+    const buffer = await readFile(`${uploadDir}/${file}`);
+    const content = JSON.parse(buffer.toString());
+    data.push(content);
+  }
   return data;
 };
 
-const deleteEmployee = (fileName: string): boolean => {
-  let deleted = false;
+const deleteEmployee = async (id: string): Promise<string | null> => {
   try {
-    fs.unlinkSync(`${uploadDir}${fileName}`);
-    deleted = true;
+    await unlink(`${uploadDir}${id}.json`);
+    return id;
   } catch (error: unknown) {
     console.error(error);
+    return null;
   }
-  return deleted;
 };
 
-const addEmployee = (fileName: string, content: string): boolean => {
-  let saved = false;
+const addEmployee = async (
+  content: Required<IEmployee>
+): Promise<string | null> => {
+  const { id } = content;
+  const stringify = JSON.stringify(content);
   try {
-    fs.writeFileSync(`${uploadDir}${fileName}`, content);
-    saved = true;
+    await writeFile(`${uploadDir}${id}.json`, stringify);
+    return id;
   } catch (error: unknown) {
     console.error(error);
+    return null;
   }
-  return saved;
 };
 
-const requestListenner = (
+const requestListenner = async (
   req: http.IncomingMessage,
   res: http.ServerResponse
 ) => {
   const { url: endpoint, method } = req;
-  if (method === "POST" && endpoint === ENDPOINTS.addEmployee) {
-    const body: Array<Uint8Array> = [];
-    req
-      .on("error", (error: Error) => console.error(error.message))
-      .on("data", (chunk: Uint8Array) => body.push(chunk))
-      .on("end", () => {
-        let parsedData: EmployeeI = JSON.parse(Buffer.concat(body).toString());
-        const isDataValid = validateEmployeeObject(parsedData);
-        if (isDataValid) {
-          if (!parsedData?.id) parsedData.id = uuid.v1();
-          const json = JSON.stringify(parsedData);
-          const savedSuccessfully = addEmployee(`${parsedData.id}.json`, json);
-          res.end(savedSuccessfully ? json : "Sorry, could not save data.");
+  const withoutParams = endpoint?.split("?")?.[0];
+  switch (withoutParams) {
+    case ENDPOINTS.employees:
```

Comment:
Why there is no method of server request?

Reply:
Oversight. Condition added